src/Parallel/NodeLock.hpp

```diff
@@ -8,66 +8,93 @@
 #include "Utilities/Gsl.hpp"
 #include "Utilities/NoSuchType.hpp"
 
+#include "Parallel/Printf.hpp"
+
 namespace Parallel {
 /*!
  * \ingroup ParallelGroup
- * \brief Create a converse CmiNodeLock
+ * \brief A typesafe wrapper for the Converse nodelock, with safe creation,
```

Comment:
I don't think that saying "the converse nodelock" is particularly helpful ... is "converse lock" a computer science concept I just can't seem to google? In any case, in this brief, I think it would be more broadly helpful to say something like "the Charm++ CmiNodeLock" or "the Charm++ node lock" instead, or any variation thereof you prefer.

I'm okay with keeping "the converse nodelock" in the details section below, but would prefer to have an extra a sentence to clarify its meaning.

Reply:
Oh, the 'converse' part is just referring to the part of the charm libraries where this is found. It is not entirely clear to me why they choose to make the distinction, but they separate these 'pseudo-global' interface pieces into a library and corresponding documentation section called Converse, and the C routines get the `Cmi...` prefix rather than the `Ck...` prefix. I'll try to clarify the documentation, and provide a link to the relevant part of the charm docs.